### ISSUE TITLE
Improve the search bar's clear button

### ISSUE THREAD

**Issue by loic-sharma** [dart-lang/pub-dev#8535]
pub.dev's search bar has an 'x' button on the right to clear the current query:

<img width="465" alt="Image" src="https://github.com/user-attachments/assets/453ca8e5-0c15-4d37-b800-a00e6cc43a10" />

Some potential improvements for this button:

1. It's hard to see in dark mode. Perhaps it should be lighter to increase the color contrast.
2. You can't focus it by pressing tab. This isn't ideal for accessibility
3. When you hover over the button, the cursor should switch to `pointer` to indicate the `x` is clickable. Currently, the cursor stays the `default` pointer cursor.
